src/extensions/__tests__/is-compatible-extension.test.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import semver from "semver";
+import {
+  isCompatibleExtension,
+} from "../extension-discovery/is-compatible-extension/is-compatible-extension";
+import type { LensExtensionManifest } from "../lens-extension";
+
+describe("Extension/App versions compatibility check", () => {
+  it("is compatible with exact version matching", () => {
+    expect(isCompatibleExtension({
+      appSemVer: semver.coerce("5.5.0"), // current app version
+    })(getExtensionManifestMock({
+      lensEngine: "5.5.0", // requested app version by extension ("semver"-format)
+    }))).toBeTruthy();
+  });
+
+  it("is compatible with higher patch-versions of main app", () => {
+    expect(isCompatibleExtension({
+      appSemVer: semver.coerce("5.5.5"), // for patch-versions
+    })(getExtensionManifestMock({
+      lensEngine: "^5.5.0",
+    }))).toBeTruthy();
+  });
```

Comment:
Can we have a test to like this but to make sure that specifying `5.5.0` is still compatible with an app version of `5.6.0`?

Reply:
There should be some "is not compatible with..." tests too.

Reply:
Probably, though I don't know how we should handle that. I looked at all the extensions I know of in the wild and they all use `^` but what should we do about extensions that use a different format? Just say that they aren't compatible? (probably I think).